packages/cadl-ranch-specs/http/spread/main.tsp

```diff
@@ -0,0 +1,158 @@
+import "@typespec/http";
+import "@azure-tools/cadl-ranch-expect";
+
+using TypeSpec.Http;
+
+@doc("Test for the spread operator.")
+@supportedBy("dpg")
+@scenarioService("/spread")
+namespace Spread;
+
+@doc("This is a simple model.")
+model ModelBodyParameter {
+  name: string;
+}
+
+@scenario
+@scenarioDoc("""
+Test case for spread named model. Should generate request body model named ModelBodyParameter.
+
+Expected request body:
+```json
+{ "name": "foo" }
+```
+""")
+@route("/model/request-body")
+@put
+op spreadModelAsRequestBody(...ModelBodyParameter): OkResponse;
+
+@doc("This is a simple model including path parameter.")
+model ModelRequestParamter {
+  @path
+  id: string;
+
+  @header
+  `x-ms-test-header`: string;
+  name: string;
+}
+
+@scenario
+@scenarioDoc("""
+Test case for spread named model with path and header parameter. 
+
+Expected path parameter: id="1"
+Expected header parameter: x-ms-test-header="bar"
+Expected request body:
+```json
+{ "name": "foo" }
+```
+""")
+@route("/model/request-parameter/{id}")
+@put
+op spreadModelAsRequestParameter(...ModelRequestParamter): OkResponse;
+
+alias AliasBodyParameter = {
+  name: string;
+};
+
+@scenario
+@scenarioDoc("""
+Test case for spread alias. 
+
+Expected request body:
+```json
+{ "name": "foo" }
+```
+""")
+@route("/alias/request-body")
+@put
+op spreadAliasAsRequestBody(...AliasBodyParameter): OkResponse;
+
+alias AliasRequestParameter = {
+  @path
+  id: string;
+
+  @header
+  `x-ms-test-header`: string;
+  name: string;
+};
+
+@scenario
+@scenarioDoc("""
+Test case for spread alias with path and header parameter. 
+
+Expected path parameter: id="1"
+Expected header parameter: x-ms-test-header="bar"
+Expected request body:
+```json
+{ "name": "foo" }
+```
+""")
+@route("/alias/request-parameter/{id}")
+@put
+op spreadAliasAsRequestParameter(...AliasRequestParameter): OkResponse;
```

Comment:
This scenario is ok, this is not a quantum model since this is an alias.

Should generate something like:
```
void spreadAliasAsRequestParameter(id: string, name: string)
```